src/coreclr/jit/flowgraph.cpp

```diff
@@ -3354,6 +3354,34 @@ void Compiler::fgCreateFunclets()
 #endif // DEBUG
 }
 
+/*-------------------------------------------------------------------------
+ *
+ * Walk the EH funclet blocks of a function to determine if the funclet
+ * section is cold. If any of the funclets are hot, then it may not be
+ * beneficial to split at fgFirstFuncletBB, moving all funclets to the
+ * cold section.
+ */
+
+bool Compiler::fgFuncletsAreCold()
+{
+    for (BasicBlock* block = fgFirstFuncletBB; block != nullptr; block = block->bbNext)
+    {
+#if HANDLER_ENTRY_MUST_BE_IN_HOT_SECTION
```

Comment:
It looks like we always define `HANDLER_ENTRY_MUST_BE_IN_HOT_SECTION` to be `1`.

Does that mean we're actually never splitting off EH any regions, or am I missing something?

Reply:
Sorry, I didn't notice `HANDLER_ENTRY_MUST_BE_IN_HOT_SECTION` is always 1. During testing, I brute-forced splitting at `fgFirstFuncletBB` via `JitStressProcedureSplitting`, so I didn't notice this condition was blocking splitting. I've removed it.

Reply:
It may not be that simple. Presumably `HANDLER_ENTRY_MUST_BE_IN_HOT_SECTION` exists for some reason. It would be good to know what that reason is.

Looking at the git blame for jit.h this define has been like this for years now. Perhaps it is no longer applicable, or perhaps it reflects some limitation in the way the jit must report EH information back to the runtime.

@jkotas do you know offhand why the jit has this constraint? If not, I can do some digging.

If it is indeed OK to change this, I'd recommend changing the define instead, so that the rest of the jit is freed from this constraint as well, and do that as a prerequisite, stand-alone change.

Reply:
I do not know the reason behind this. The EE side is using [`GetCodeAddressForRelOffset`](https://github.com/dotnet/runtime/search?q=GetCodeAddressForRelOffset) method to resolve the handler offset to IP. This method deals with handler entrypoints in the cold section correctly.